### ISSUE TITLE
collectResult "result is not iterable" when recursing non-lit custom elements

### ISSUE THREAD

**Issue by btopro** [lit/lit#4711]
### Should this be an RFC?

- [X] This is not a substantial change

### Which package is this a feature request for?

SSR (@lit-labs/ssr)

### Description

We're trying to hydrate content for a CMS. Users store a mix of HTML and Custom Elements in static files, these files are then loaded by our CMS relative to the route. Ideally, we'd SSR hydrate them prior to what's sent, but store them unhydrated. While conditionally supporting `globalThis.document` and other such missing classes in global node environment, I noticed that if I dynamically loaded definitions that in their `import()` tree ran into any elements that were `HTMLElement`, that entire part of the template would brick.

I am not asking for full `HTMLElement` support, just a graceful serving of the parts of the template that *did* successfully render. 5 elements correctly rendering their media to the screen would still be better than 0 elements in a response (as happens currently).

- Example code: https://github.com/haxtheweb/webcomponents/blob/master/api/hydrateSsr.js
- Endpoint running code: https://webcomponents.hax.cloud/api/hydrateSsr?type=link&q=https://oer.hax.psu.edu/ajh6037/sites/chem110/pages/lesson-1/molecular-polarity/index.html?1721834563
- HTML file in question: https://oer.hax.psu.edu/ajh6037/sites/chem110/pages/lesson-1/molecular-polarity/index.html?1721834563
- Actual rendered in context (for reference): https://oer.hax.psu.edu/ajh6037/sites/chem110/lesson-1/molecular-polarity

This is currently powered using a small patch (to be issued in the PR) - https://github.com/haxtheweb/webcomponents/blob/master/patches/%40lit-labs%2Bssr%2B3.2.2.patch

### Alternatives and Workarounds

if `collectResult` had a catch to ensure that it is not `undefined` (thrown when a custom element definition fails to load properly) then it will deliver the template in these scenarios. While not a perfect solution, this work around allows a template to at least be delivered as opposed to failing to deliver because of a type error.

```js
/**
 * Joins a RenderResult into a string
 */
export const collectResult = async (result) => {
    let value = '';
    if (result) {
      for (const chunk of result) {
        value +=
              typeof chunk === 'string' ? chunk : await collectResult(await chunk);
      }
    }
    return value;
};
```

**Comment by justinfagnani** [comment]
I'm having a little trouble parsing the problem here. collectResult should not depend on whether any elements are defined or built with Lit. Can you summarize things again?
